### DISCUSSION TITLE
Programmatic usage?

### DISCUSSION THREAD

**Discussion by heftymouse** [git-ecosystem/git-credential-manager#1296] [Q&A] [answered]
Is there a way to use GCM as a library in a .NET app? We're using libgit2sharp and it'd be good to take advantage for existing support for authentication as well.

**Comment by ldennington** [comment, marked as answer]
While we do not have a library version of GCM available, libgit2sharp should allow you to set a `CredentialsHandler` when you're doing a fetch or other operation. In that callback, you can always shell out to GCM calling `get` to acquire a credential and either `store` or `erase` after the operation either succeeds or fails.
